source/client-side-encryption/tests/README.rst

```diff
@@ -1080,3 +1161,283 @@ Invalid Hostname in KMS Certificate
+
+      {
+            "aws": {
+               "accessKeyId": <set from environment>,
+               "secretAccessKey": <set from environment>
+            },
+            "azure": {
+               "tenantId": <set from environment>,
+               "clientId": <set from environment>,
+               "clientSecret": <set from environment>,
+               "identityPlatformEndpoint": "127.0.0.1:8002"
+            },
+            "gcp": {
+               "email": <set from environment>,
+               "privateKey": <set from environment>,
+               "endpoint": "127.0.0.1:8002"
+            },
+            "kmip" {
+               "endpoint": "127.0.0.1:5698"
+            }
+      }
+
+2. Create a ``ClientEncryption`` object named ``client_encryption_with_tls`` with the following KMS providers:
+
+   .. code:: javascript
+
+      {
+            "aws": {
+               "accessKeyId": <set from environment>,
+               "secretAccessKey": <set from environment>
+            },
+            "azure": {
+               "tenantId": <set from environment>,
+               "clientId": <set from environment>,
+               "clientSecret": <set from environment>,
+               "identityPlatformEndpoint": "127.0.0.1:8002"
+            },
+            "gcp": {
+               "email": <set from environment>,
+               "privateKey": <set from environment>,
+               "endpoint": "127.0.0.1:8002"
+            },
+            "kmip" {
+               "endpoint": "127.0.0.1:5698"
+            }
+      }
+
+   Add TLS options for the ``aws``, ``azure``, ``gcp``, and
+   ``kmip`` providers to use the following options:
+
+   - ``tlsCAFile`` (or equivalent) set to `ca.pem`_. This MAY be configured system-wide. 
+   - ``tlsCertificateKeyFile`` (or equivalent) set to `client.pem`_
+
+3. Create a ``ClientEncryption`` object named ``client_encryption_expired`` with the following KMS providers:
+
+   .. code:: javascript
+
+      {
+            "aws": {
+               "accessKeyId": <set from environment>,
+               "secretAccessKey": <set from environment>
+            },
+            "azure": {
+               "tenantId": <set from environment>,
+               "clientId": <set from environment>,
+               "clientSecret": <set from environment>,
+               "identityPlatformEndpoint": "127.0.0.1:8000"
+            },
+            "gcp": {
+               "email": <set from environment>,
+               "privateKey": <set from environment>,
+               "endpoint": "127.0.0.1:8000"
+            },
+            "kmip" {
+               "endpoint": "127.0.0.1:8000"
+            }
+      }
+
+   Add TLS options for the ``aws``, ``azure``, ``gcp``, and
+   ``kmip`` providers to use the following options:
+
+   - ``tlsCAFile`` (or equivalent) set to `ca.pem`_. This MAY be configured system-wide.
+
+4. Create a ``ClientEncryption`` object named ``client_encryption_invalid_hostname`` with the following KMS providers:
```

Comment:
The two new test cases make the `KMS TLS Tests` redundant. I can remove them in a follow-up PR.

Reply:
Draft PR: https://github.com/mongodb/specifications/pull/1093